src/main/java/net/justonedev/mc/clickthrough/ChestEvents.java

```diff
@@ -71,6 +73,12 @@ private boolean canOpenContainer(Block container, BlockInventoryHolder holder) {
                 Block otherHalf = findOtherChestHalf(container, doubleChest);
                 if (otherHalf != null && otherHalf.getRelative(BlockFace.UP).getType().isOccluding()) return false;
             }
+
+            if (container.getType().toString().endsWith("SHULKER_BOX")) {
```

Comment:
Instead of comparing strings, you can combine this line and the next by just checking if it's a Shulker Inventory Holder and assigning it a variable (ShulkerBox extends BlockState).

```suggestion
            if (holder instanceof ShulkerBox shulkerBox) {
```